### PR TITLE
Move unary operators around shape transformation

### PR DESCRIPTION
**Pull request by pfultz2**
This also removes the old unary rewriting in simplify_reshapes. They are not really useful.

### CODE REVIEW

File: src/simplify_reshapes.cpp [hunk 1]
```diff
@@ -645,41 +645,106 @@ struct find_reshape_cont
...
+
+        bool move_up   = can_fuse_unary(x);
+        bool move_down = can_fuse_unary(y);
+
+        if(move_up and move_down)
+        {
+            if(x->name() == "@literal")
+                move_down = false; // NOLINT(bugprone-branch-clone)
+            else if(yops.empty())
+                move_up = false;
```
Comment (TedThemistokleous) [inline src/simplify_reshapes.cpp:727]: Add coverage for these cases

Comment (pfultz2) [reply src/simplify_reshapes.cpp:727]: Added tests.

**Review by TedThemistokleous — APPROVED** [review]

File: src/simplify_reshapes.cpp [hunk 2]
```diff
@@ -645,41 +645,108 @@ struct find_reshape_cont
...
     auto matcher() const
     {
-        return pointwise(match::used_once(),
-                         match::nargs(1),
-                         match::args(match_transpose_contiguous_reshaper()));
+        auto output_not_pointwise =
+            match::none_of(match::skip_output(match::name("contiguous"))(match::pointwise()));
+        auto input_has_shape_transform =
+            match::args(match::skip(match::name("contiguous"))(match::name(shape_transforms())));
+        return match::pointwise(
```
Comment (shivadbhavsar) [inline src/simplify_reshapes.cpp:669]: shouldn't this check for unary? ie. `match::nargs(1)`

Comment (shivadbhavsar) [reply src/simplify_reshapes.cpp:669]: nvm, all the accepted input ops are unary I see

**Review by shivadbhavsar — APPROVED** [review]

**Pull request merged**